File: Java/Tema5_Ejercicio4.java
/*
 * TEMA 5: ARREGLOS Y LISTAS - EJERCICIO 4
 * ========================================
 * Escriba un programa que ordene un arreglo en orden ascendente.
 */

import java.util.Arrays;

public class Tema5_Ejercicio4 {
    public static void main(String[] args) {
        int[] numeros = {64, 34, 25, 12, 22, 11, 90, 33, 55, 44};
        
        System.out.println("=== ARREGLO ORIGINAL ===");
        System.out.println(Arrays.toString(numeros));
        
        // Ordenamiento de burbuja (bubble sort)
        for (int i = 0; i < numeros.length - 1; i++) {
            for (int j = 0; j < numeros.length - 1 - i; j++) {
                if (numeros[j] > numeros[j + 1]) {
                    // Intercambiar
                    int temp = numeros[j];
                    numeros[j] = numeros[j + 1];
                    numeros[j + 1] = temp;
                }
            }
        }
        
        System.out.println("\n=== ARREGLO ORDENADO ===");
        System.out.println(Arrays.toString(numeros));
        
        // Forma rápida usando Arrays.sort()
        int[] numeros2 = {64, 34, 25, 12, 22, 11, 90, 33, 55, 44};
        Arrays.sort(numeros2);
        System.out.println("\n=== USANDO Arrays.sort() ===");
        System.out.println(Arrays.toString(numeros2));
    }
}

/*
 * SALIDA ESPERADA:
 * === ARREGLO ORIGINAL ===
 * [64, 34, 25, 12, 22, 11, 90, 33, 55, 44]
 * 
 * === ARREGLO ORDENADO ===
 * [11, 12, 22, 25, 33, 34, 44, 55, 64, 90]
 * 
 * === USANDO Arrays.sort() ===
 * [11, 12, 22, 25, 33, 34, 44, 55, 64, 90]
 */

File: JS/Tema5_Ejercicio4.js
/*
 * TEMA 5: ARREGLOS Y LISTAS - EJERCICIO 4
 * ========================================
 * Implementar ordenamiento (bubble sort y sort()).
 * 
 * EXPLICACIÓN:
 * - Bubble sort: algoritmo simple de ordenamiento
 * - Array.sort(): método nativo
 * - Comparación de métodos
 */

function bubbleSort(arr) {
    const copia = [...arr];
    for (let i = 0; i < copia.length; i++) {
        for (let j = 0; j < copia.length - i - 1; j++) {
            if (copia[j] > copia[j + 1]) {
                [copia[j], copia[j + 1]] = [copia[j + 1], copia[j]];
            }
        }
    }
    return copia;
}

const numeros = [64, 34, 25, 12, 22, 11, 90, 88];

console.log("=== ORDENAMIENTO DE ARRAYS ===");
console.log(`Array original: ${numeros}`);

const ordenadoBubble = bubbleSort(numeros);
console.log(`\nBubble sort: ${ordenadoBubble}`);

const ordenadoSort = [...numeros].sort((a, b) => a - b);
console.log(`Array.sort(): ${ordenadoSort}`);

// Ordenamiento descendente
const descendente = [...numeros].sort((a, b) => b - a);
console.log(`Descendente: ${descendente}`);

/*
 * SALIDA ESPERADA:
 * === ORDENAMIENTO DE ARRAYS ===
 * Array original: 64,34,25,12,22,11,90,88
 * 
 * Bubble sort: 11,12,22,25,34,64,88,90
 * Array.sort(): 11,12,22,25,34,64,88,90
 * Descendente: 90,88,64,34,25,22,12,11
 */
